platforms/jvm/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/RtxInstallationSupplier.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.api.Transformer;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.util.Collections;
+import java.util.Set;
+
+public class RtxInstallationSupplier extends AutoDetectingInstallationSupplier {
+
+    private final Provider<String> rtxDataDir;
+    private final Provider<String> rtxUserHome;
+
+    @Inject
+    public RtxInstallationSupplier(ProviderFactory factory) {
+        super(factory);
+        rtxDataDir = getEnvironmentProperty("RTX_DATA_DIR");
+        rtxUserHome = getEnvironmentProperty("XDG_DATA_HOME")
+            .orElse(getSystemProperty("user.home")
```

Comment:
Does rtx happen to use [`directories.data_dir()`](https://crates.io/crates/directories) indirectly or anything? (I don't see it in Cargo.toml) since that would have some special handling for both Windows and Mac locations too.

Gradle's autodetection should follow whatever rtx does, so this is just a sanity check that this is rtx's behavior even in Windows/OSX contexts

Reply:
rtx has the same, simple, logic: https://github.com/jdx/rtx/blob/main/src/env.rs#L24

windows isn't supported for rtx (yet, if ever)

Reply:
Ok this can be resolved. (I can't do it)

Reply:
it was a good question! I had to check the source to make sure